packages/slate-plugins/src/element/types.ts

```diff
@@ -7,8 +7,18 @@ export interface ToggleBlockEditor extends Editor {
 }
 
 export interface GetRenderElementOptions {
+  /**
+   * Type of the element
+   */
   type: string;
+  /**
+   * React component to render the element
+   */
   component: any;
+  /**
+   * Options passed to the component as props
+   */
+  [key: string]: any;
```

Comment:
with the above mentioned pattern this could be omitted, and having custom interfaces for the specific elements, and where used, TypeScript could infer which parameters exist, based on the type.

Reply:
Going to update that for all the plugins then.